src/library_manager/lib_manager.c

```diff
@@ -47,59 +47,86 @@ struct lib_manager_dma_ext {
 
 static struct ext_library loader_ext_lib;
 
+struct lib_manager_mod_ctx {
+	struct sof_man_fw_desc *desc;
+	size_t segment_size[3];
+};
+
+static struct lib_manager_mod_ctx *lib_manager_get_mod_ctx(int module_id)
+{
+	uint32_t lib_id = LIB_MANAGER_GET_LIB_ID(module_id);
+	struct ext_library *_ext_lib = ext_lib_get();
+
+	return _ext_lib->desc[lib_id];
+}
+
+struct sof_man_fw_desc *lib_manager_get_library_module_desc(int module_id)
+{
+	struct lib_manager_mod_ctx *ctx = lib_manager_get_mod_ctx(module_id);
+	uint8_t *buffptr = (uint8_t *)(ctx ? ctx->desc : NULL);
```

Comment:
superfluous double checks. Why not just `if (!ctx) return NULL;`?

Reply:
@softwarecki I wouldn't really call this superfluous. An alternative would look like
```
if (!ctx || !ctx->desc)
	return NULL;
uint8_t *buffptr = ctx->desc;
```
so you anyway have to check up to two conditions - just as well as now. I'd say it's a matter of preference